File: intellij-lsp/src/com/github/gtache/lsp/settings/server/parser/ConfigType.java
package com.github.gtache.lsp.settings.server.parser;

import org.jetbrains.annotations.Nullable;

public enum ConfigType {
    FLAT, JSON, XML;

    @Nullable
    public static ConfigType forExt(final String ext) {
        if ("FLAT".equalsIgnoreCase(ext)) {
            return FLAT;
        } else if ("JSON".equalsIgnoreCase(ext)) {
            return JSON;
        } else if ("XML".equalsIgnoreCase(ext)) {
            return XML;
        } else return null;
    }

    public static String toExt(final ConfigType typ) {
        if (typ == FLAT) {
            return "txt";
        } else if (typ == JSON) {
            return "json";
        } else if (typ == XML) {
            return "xml";
        } else return null;
    }
}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/settings/server/parser/ConfigType.kt
package com.github.gtache.lsp.settings.server.parser

/**
 * Configuration type
 */
enum class ConfigType {
    FLAT, JSON, XML;

    companion object {
        /**
         * Returns a type given an [extension]
         */
        fun fromExtension(extension: String?): ConfigType? {
            return if ("txt".equals(extension, ignoreCase = true) || "ini".equals(extension, ignoreCase = true)) {
                FLAT
            } else if (JSON.name.equals(extension, ignoreCase = true)) {
                JSON
            } else if (XML.name.equals(extension, ignoreCase = true)) {
                XML
            } else null
        }

        /**
         * Returns an extension given a [type]
         */
        fun toExtension(type: ConfigType): String? {
            return when (type) {
                FLAT -> "txt"
                JSON -> JSON.name.lowercase()
                XML -> XML.name.lowercase()
                else -> null
            }
        }
    }
}